libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -336,7 +336,11 @@ void UARTDriver::_uart_start_connection(void)
 
     // set baudrate
     tcgetattr(_fd, &t);
-    cfsetspeed(&t, _uart_baudrate);
+#if defined(__ANDROID__) && (__BORLANDC__)
```

Comment:
it's missing in Android NDK, but i included and fix it with this PR https://github.com/ArduPilot/ardupilot/pull/5394.